utils/user_info.py

```diff
@@ -74,6 +76,32 @@ def has_p_ownership(curr_user, curr_problem):
     is_owner = (curr_user.username == curr_problem.owner.username)
     return is_owner
 
+def has_p_auth(user, problem):
```

Comment:
Could you please rename this function to has_problem_authentication or another name?
Avoiding shortened name in our code can improve the quality of code